fork-visualizer/src/app/components/NodeTracker.jsx

```diff
@@ -0,0 +1,119 @@
+"use client"
+
+import { ApiPromise, WsProvider } from "@polkadot/api";
+import { useState } from "react"
+
+
+const defaultFunc = () => console.log("oops this is the default func")
+export function NodeState ({ wsAddress, updateStuff, removeNode }) {
+
+    const [ isSubscribed, setIsSubscribed ] = useState(false)
+    const [ loading, setLoading ] = useState(false)
+    const [ unsubscribeFromNode, setUnsubscribeFromNode ] = useState(() => defaultFunc)
+    const [ latestHash, setLatestHash ] = useState("")
+
+    const subscribe = async () => {
+        if (isSubscribed) return
+
+        setLoading(true)
+
+        console.log("subscribing...", wsAddress)
```

Comment:
is this needed?

Reply:
no I can remove it, it was mostly used for debugging